runtime-modules/project-token/src/types.rs

```diff
@@ -76,42 +137,325 @@ pub enum TransferPolicy<Hash> {
     Permissioned(Hash),
 }
 
+// TransferPolicyParams => TransferPolicy conversion
+impl<Hash, SingleDataObjectUploadParams>
+    From<TransferPolicyParams<WhitelistParams<Hash, SingleDataObjectUploadParams>>>
+    for TransferPolicy<Hash>
+{
+    fn from(
+        params: TransferPolicyParams<WhitelistParams<Hash, SingleDataObjectUploadParams>>,
+    ) -> Self {
+        match params {
+            TransferPolicyParams::Permissioned(whitelist_params) => {
+                Self::Permissioned(whitelist_params.commitment)
+            }
+            TransferPolicyParams::Permissionless => Self::Permissionless,
+        }
+    }
+}
+
 impl<Hash> Default for TransferPolicy<Hash> {
     fn default() -> Self {
         TransferPolicy::<Hash>::Permissionless
     }
 }
 
-/// The possible offering variants: This is a stub
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, Default)]
+pub struct VestingScheduleParams<BlockNumber> {
+    // Vesting duration
+    pub(crate) duration: BlockNumber,
+    // Number of blocks before the linear vesting begins
+    pub(crate) blocks_before_cliff: BlockNumber,
+    // Initial, instantly vested amount once linear vesting begins (percentage of total amount)
+    pub(crate) cliff_amount_percentage: Permill,
+}
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, Default)]
+pub struct VestingSchedule<BlockNumber, Balance> {
+    // Block at which the vesting begins
+    pub(crate) start_block: BlockNumber,
+    // Linear vesting duration
+    pub(crate) duration: BlockNumber,
```

Comment:
lets rename this to more explicitly be `linear_vesting_duration`.

Reply:
Renamed in https://github.com/Joystream/joystream/pull/3618/commits/d8e70c04967c6c8050a12b204be1f39974e955ef